projects/user_publication/utils.py

```diff
@@ -27,10 +27,15 @@
 ]
 
 
-def add_to_all_sources(pub, source):
-    LOG.info(f"Publication already exists - {pub.title} - updating other source")
-    setattr(pub, source, True)
-    pub.save()
+def add_source_to_pub(pub, source):
+    LOG.info(
+        f"Publication already exists - {pub.title}"
+        f" - adding other source - {source} - This check might be outdated"
+    )
+    source = pub.source.get_or_create(name=source)[0]
+    source.is_source = True
```

Comment:
Is this ever false?

Reply:
yes. When we only get citations count for a publication but are not able to find that publication referencing chameleon when we search in that source.